### PR TITLE
Add Microsoft.AspNetCore.Mvc.Razor.Extensions

### PR DESCRIPTION
**Pull request by ryanbrandenburg**
A port of `Microsoft.AspNetCore.Mvc.Razor.Host`.

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.Razor.Extensions/Internal/DesignTimeRazorPathNormalizer.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+
+namespace Microsoft.AspNetCore.Razor.Internal
+{
+    public class DesignTimeRazorPathNormalizer : RazorPathNormalizer
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Razor.Extensions/Internal/DesignTimeRazorPathNormalizer.cs]: Where is this guy used? I don't know if `RazorPathNormalizer` still exists

File: src/Microsoft.AspNetCore.Mvc.Razor.Extensions/Internal/RazorPathNormalizer.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Razor.Internal
+{
+    public class RazorPathNormalizer
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Razor.Extensions/Internal/RazorPathNormalizer.cs]: Whatsit?

File: src/Microsoft.AspNetCore.Mvc.Razor.Extensions/Microsoft.AspNetCore.Mvc.Razor.Extensions.csproj [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    <ProjectReference Include="../Microsoft.AspNetCore.Razor.Evolution/Microsoft.AspNetCore.Razor.Evolution.csproj" />
+    
+    <PackageReference Include="Microsoft.Extensions.Caching.Memory" Version="$(AspNetCoreVersion)" />
+    <PackageReference Include="Microsoft.Extensions.FileProviders.Physical" Version="$(AspNetCoreVersion)" />
+    <PackageReference Include="Microsoft.Extensions.PropertyHelper.Sources" Version="$(AspNetCoreVersion)" PrivateAssets="All" />
+  </ItemGroup>
+
+  <ItemGroup Condition=" '$(TargetFramework)' == 'netstandard1.6' ">
+    <PackageReference Include="System.ComponentModel.TypeConverter" Version="$(CoreFxVersion)" />
+  </ItemGroup>
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Razor.Extensions/Microsoft.AspNetCore.Mvc.Razor.Extensions.csproj]: Please prune this down to the minimum set of dependencies. Cache and FileProvider should not be used here

File: src/Microsoft.AspNetCore.Mvc.Razor.Extensions/baseline.net45.json [hunk 4]
```diff
@@ -0,0 +1,1689 @@
+{
+  "AssemblyIdentity": "Microsoft.AspNetCore.Mvc.Razor.Extensions, Version=1.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60",
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Razor.Extensions/baseline.net45.json]: We don't need these baselines files. There is no compatibility bar between this and the old assembly.

File: test/Microsoft.AspNetCore.Mvc.Razor.Extensions.Test/Internal/SolutionPathUtility.cs [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+using System;
+using System.IO;
+using System.Reflection;
+using Microsoft.Extensions.PlatformAbstractions;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Extensions
+{
+    public static class SolutionPathUtility
+    {
+        private const string SolutionName = "Mvc.sln";
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Mvc.Razor.Extensions.Test/Internal/SolutionPathUtility.cs]: This isn't right.

File: test/Microsoft.AspNetCore.Mvc.Razor.Extensions.Test/LineMappingsSerializer.cs [hunk 6]
```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Text;
+using Microsoft.AspNetCore.Razor.Evolution;
+
+namespace Microsoft.AspNetCore.Razor.Extensions
+{
+    public static class LineMappingsSerializer
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Mvc.Razor.Extensions.Test/LineMappingsSerializer.cs]: There's a follow up item here to dedupe this test infrastructure.

File: src/Microsoft.AspNetCore.Mvc.Razor.Extensions/ModelDirective.cs [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Razor.Evolution;
+using Microsoft.AspNetCore.Razor.Evolution.Intermediate;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Extensions
+{
+    public static class ModelDirective
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Mvc.Razor.Extensions/ModelDirective.cs]: Shouldn't act on this now but thought I'd bring it up. @rynowak / all what do you think about making `@model` part of core Razor?

File: test/Microsoft.AspNetCore.Mvc.Razor.Extensions.Test/LineMappingsSerializer.cs (hunk 6 above)
Comment (NTaylorMullen) [inline test/Microsoft.AspNetCore.Mvc.Razor.Extensions.Test/LineMappingsSerializer.cs]: Yup, https://github.com/aspnet/Razor/issues/991. Should update that guy to reflect the fact that we no longer need a package (all codegen tests now live in Razor). We do however, need a separate infrastructure project

File: src/Microsoft.AspNetCore.Mvc.Razor.Extensions/ModelDirective.cs (hunk 7 above)
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Razor.Extensions/ModelDirective.cs]: What would it do?

File: src/Microsoft.AspNetCore.Mvc.Razor.Extensions/InjectDirectiveTargetExtension.cs [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Razor.Evolution.CodeGeneration;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Extensions
+{
+    public class InjectDirectiveTargetExtension : IInjectDirectiveTargetExtension
+    {
+        private const string RazorInjectAttribute = "[global::Microsoft.AspNetCore.Razor.Internal.RazorInjectAttribute]";
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Razor.Extensions/InjectDirectiveTargetExtension.cs]: IMO no

File: src/Microsoft.AspNetCore.Mvc.Razor.Extensions/ModelDirective.cs (hunk 7 above)
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Mvc.Razor.Extensions/ModelDirective.cs]: It'd change the classes base type to take a generic of TModel

**Comment by ryanbrandenburg** [comment]
:up::date:

File: src/Microsoft.AspNetCore.Mvc.Razor.Extensions/ModelDirective.cs (hunk 7 above)
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Razor.Extensions/ModelDirective.cs]: ehhhhhhhhhhhhhhhhhhhhhhhhhhhhhh

**Pull request merged**